samples/tests/traits/mod.rs

```diff
@@ -0,0 +1,43 @@
+samples::test! {
+    volumetric;
+    /// ANCHOR: volumetric
+    trait Volumetric {
+        /// Strain energy density
+        fn psi(&self, j: f64) -> f64;
+        /// Derivative of strain energy with respect to $J$
+        fn d_psi(&self, j: f64, b_psi: f64) -> (f64, f64);
+        /// The volumetric contribution to the second Piola-Kirchhoff stress is
+        /// a scalar multiplied by $C^{-1}$ where $C = I + 2E$ in terms of the
+        /// Green-Lagrange strain $E$. The derivative of $J$ with respect to $E$
+        /// is $J C^{-1}$. We'll call the volumetric contribution that scalar
+        /// multiple of $C^{-1}$.
+        fn stress(&self, j: f64) -> f64 {
+            let (_, d_psi) = self.d_psi(j, 1.0);
+            d_psi * j
+        }
+    }
+
+    struct Ogden {
+        k: f64,
+    }
+    impl Ogden {
+        pub fn stress_analytic(&self, j: f64) -> f64 {
+            self.k * 0.5 * (j * j - 1.0)
+        }
+    }
+    impl Volumetric for Ogden {
+        #[autodiff(d_psi, Reverse, Active, Active)]
+        fn psi(&self, j: f64) -> f64 {
+            self.k * 0.25 * (j * j - 1.0 - 2.0 * j.ln())
```

Comment:
There are many such models that one would like to implement. This one is $\beta=-2$ of Eq 8, but you can see many other models in this paper: https://doi.org/10.1016/j.ijsolstr.2020.01.019

Reply:
this works, so please feel free to add more tests.